File: app/src/main/java/com/example/supisara/coinconverter/ConvertNewActivity.java
package com.example.supisara.coinconverter;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.InputType;
import android.view.View;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by supisara on 5/24/2018 AD.
 */

public class ConvertNewActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.convertnew_activity);
    }

    public void convertButtonClicked(View view){
        EditText inputEditText = (EditText) findViewById(R.id.inputEditText);
        inputEditText.setInputType(InputType.TYPE_CLASS_NUMBER);
        String inputText = inputEditText.getText().toString();

        EditText rateEditText = (EditText) findViewById(R.id.rateEditText);
        rateEditText.setInputType(InputType.TYPE_CLASS_NUMBER);
        String rateText = rateEditText.getText().toString();

        double input = 0, rate = 0;

        if(inputText.matches("[0-9]+(\\.[0-9]+)?") && rateText.matches("[0-9]+(\\.[0-9]+)?")) {
            input = Integer.parseInt(inputEditText.getText().toString());
            rate = Integer.parseInt(inputEditText.getText().toString());
        }

        //print the converted result
        TextView outputTextView = (TextView) findViewById(R.id.outputTextView);
        outputTextView.setInputType(InputType.TYPE_CLASS_NUMBER);

        outputTextView.setText(String.valueOf(convert(input, rate)));

    }

    public double convert(double number, double convertUnit){
        return number * convertUnit;
    }

    public void clearButtonClicked(View view){
        EditText inputEditText = (EditText) findViewById(R.id.inputEditText);
        inputEditText.setInputType(InputType.TYPE_CLASS_NUMBER);
        inputEditText.setText("");

        TextView outputTextView = (TextView) findViewById(R.id.outputTextView);
        outputTextView.setInputType(InputType.TYPE_CLASS_NUMBER);
        outputTextView.setText("");

        EditText rateEditText = (EditText) findViewById(R.id.rateEditText);
        rateEditText.setInputType(InputType.TYPE_CLASS_NUMBER);
        rateEditText.setText("");
    }
}

File: app/src/main/java/com/example/supisara/coinconverter/ConvertNewActivity.kt
package com.example.supisara.coinconverter

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.InputType
import android.view.View
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView

import java.util.ArrayList

/**
 * Created by supisara on 5/24/2018 AD.
 */

class ConvertNewActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.convertnew_activity)
    }

    fun convertButtonClicked(view: View) {
        val inputEditText = findViewById<View>(R.id.inputEditText) as EditText
        inputEditText.inputType = InputType.TYPE_CLASS_NUMBER
        val inputText = inputEditText.text.toString()

        val rateEditText = findViewById<View>(R.id.rateEditText) as EditText
        rateEditText.inputType = InputType.TYPE_CLASS_NUMBER
        val rateText = rateEditText.text.toString()

        val outputTextView = findViewById<View>(R.id.outputTextView) as TextView
        outputTextView.inputType = InputType.TYPE_CLASS_NUMBER

        if(inputText != null && rateText != null && inputText != "-" && rateText != "-" && inputText != "" && rateText != ""){
            var input = 0.0
            var rate = 0.0
            input = Integer.parseInt(inputEditText.text.toString()).toDouble()
            rate = Integer.parseInt(inputEditText.text.toString()).toDouble()
            outputTextView.text = convert(input, rate).toString()
        } else{
            outputTextView.text = "Please enter a number"

        }
    }

    fun convert(number: Double, convertUnit: Double): Double {
        return number * convertUnit
    }

    fun clearButtonClicked(view: View) {
        val inputEditText = findViewById<View>(R.id.inputEditText) as EditText
        inputEditText.inputType = InputType.TYPE_CLASS_NUMBER
        inputEditText.setText("")

        val outputTextView = findViewById<View>(R.id.outputTextView) as TextView
        outputTextView.inputType = InputType.TYPE_CLASS_NUMBER
        outputTextView.text = ""

        val rateEditText = findViewById<View>(R.id.rateEditText) as EditText
        rateEditText.inputType = InputType.TYPE_CLASS_NUMBER
        rateEditText.setText("")
    }
}
